### PR TITLE
Require the cl_mipmap_filter_mode_img type for cl_img_generate_mipmap.

### PR DESCRIPTION
**Pull request by zfigura**
To help tools which parse the cl.xml file and conditionally output types.

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
LGTM, but it'd be great to get a +1 from @Jeremy-Kemp before merging.  Thanks!

**Review by Jeremy-Kemp — APPROVED** [review]
Looks good, thank you!

**Pull request merged**
